src/abstractdataframe/abstractdataframe.jl

```diff
@@ -67,22 +67,47 @@ _names(df::AbstractDataFrame) = _names(index(df))
 
 Compat.hasproperty(df::AbstractDataFrame, s::Symbol) = haskey(index(df), s)
 
-function rename!(df::AbstractDataFrame, vals::AbstractVector{Symbol}; makeunique::Bool=false)
+function rename!(df::AbstractDataFrame, vals::AbstractVector{Symbol};
+                 makeunique::Bool=false)
     rename!(index(df), vals, makeunique=makeunique)
     return df
 end
 
-function rename!(df::AbstractDataFrame, args...)
-    rename!(index(df), args...)
+function rename!(df::AbstractDataFrame, vals::AbstractVector{<:AbstractString};
+                 makeunique::Bool=false)
+    rename!(index(df), Symbol.(vals), makeunique=makeunique)
     return df
 end
+
+function rename!(df::AbstractDataFrame,
+                 args::Union{AbstractVector{<:Pair}, AbstractDict})
```

Comment:
Why not do this to avoid having to check manually?
```suggestion
                 args::Union{AbstractVector{<:Pair{Union{Symbol,String}, Union{Symbol,String}}, AbstractDict})
```

Reply:
The reason is type promotion rules:
```
julia> ["a"=>"b", :c=>:d]
2-element Array{Pair{Any,Any},1}:
 "a" => "b"
  :c => :d
```
and you would have to write manually:
```
julia> Pair{Union{Symbol,String}, Union{Symbol,String}}["a"=>"b", :c=>:d]
2-element Array{Pair{Union{String, Symbol},Union{String, Symbol}},1}:
 "a" => "b"
  :c => :d
```
which I do not think anyone would do in practice.

Reply:
I see. But do we really want to allow mixing these?

Reply:
I can disallow them, but `Pair{Union{Symbol,String}, Union{Symbol,String}}` actually allows mixing.

So do we want to be stricter and actually only allow `Pair{Union{T}, Union{U}} where {T<:Union{Symbol, AbstractString}, U<:Union{Symbol, AbstractString}}` (so essentially that `first` and `last` are homogeneous).

Or we go for `Pair{Union{Symbol,String}, Union{Symbol,String}}` signature and technically allow mixing but in practice disallow it due to the problem I have described above (this would be simplest to change in the code).

Reply:
I don't have a strong opinion, but being strict would probably be OK.

Reply:
```suggestion
                 args::Union{AbstractVector{<:Pair{Union{Symbol,AbstractString}, Union{Symbol,AbstractString}}, AbstractDict})
```

Reply:
OK - I will change it to be strict then.